core/test/usLDAPQueryTest.cpp

```diff
@@ -0,0 +1,123 @@
+/*=============================================================================
+ 
+ Library: CppMicroServices
+ 
+ Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+ file at the top-level directory of this distribution and at
+ https://github.com/saschazelzer/CppMicroServices/COPYRIGHT .
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ 
+ =============================================================================*/
+
+#include <usFrameworkFactory.h>
+#include <usFramework.h>
+
+#include <usBundle.h>
+#include <usLDAPFilter.h>
+#include <usLog.h>
+
+#include "usTestUtils.h"
+#include "usTestingMacros.h"
+
+using namespace us;
+
+void TestLDAPQueryBundleMetaData(BundleContext* frameworkCtx)
+{
+    std::vector<std::shared_ptr<Bundle>> bundles = frameworkCtx->GetBundles();
+    LDAPFilter ldap( "(bundle.testproperty=YES)" );
+    
+    int actualMatchCount = 0;
+    int expectedMatchCount = 0;
+    
+    std::string bundleName;
+    Any bundleTestProperty;
+    
+    for (std::shared_ptr<Bundle> bundle : bundles)
```

Comment:
I'd suggest using a ```const auto&``` for the range declaration to leverage type deduction and maintain const correctness. The ```bundle``` object will not be modified in the loop and ```LDAPFilter::Match(const Bundle& )``` expects a ```const&```.